plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergBucketing.java

```diff
@@ -0,0 +1,238 @@
+        assertMaxIntBucketEquals(icebergType, value, expectedMaxIntBucket);
+    }
+
+    private void assertBucketEquals(Type icebergType, Object value)
+    {
+        assertBucketNumberEquals(icebergType, value, Integer.MAX_VALUE);
+
+        assertBucketNumberEquals(icebergType, value, 2);
+        assertBucketNumberEquals(icebergType, value, 7);
+        assertBucketNumberEquals(icebergType, value, 31);
+        assertBucketNumberEquals(icebergType, value, 32);
+        assertBucketNumberEquals(icebergType, value, 100);
+        assertBucketNumberEquals(icebergType, value, 10000);
+        assertBucketNumberEquals(icebergType, value, 524287); // prime number
+        assertBucketNumberEquals(icebergType, value, 1 << 30);
+    }
+
+    private void assertBucketNumberEquals(Type icebergType, Object value, int bucketCount)
+    {
+        Integer icebergBucket = computeIcebergBucket(icebergType, value, bucketCount);
+        Integer trinoBucket = computeTrinoBucket(icebergType, value, bucketCount);
+
+        assertEquals(
+                icebergBucket,
+                trinoBucket,
+                format("icebergType=%s, icebergBucket=%d, trinoBucket=%d", icebergType, icebergBucket, trinoBucket));
+    }
+
+    private void assertMaxIntBucketEquals(Type icebergType, Object value, Integer expectedMaxIntBucket)
+    {
+        Integer icebergBucketHash = computeIcebergBucket(icebergType, value, Integer.MAX_VALUE);
+        Integer trinoBucketHash = computeTrinoBucket(icebergType, value, Integer.MAX_VALUE);
+
+        // Ensure hash is stable and does not change
+        assertEquals(
+                icebergBucketHash,
+                expectedMaxIntBucket,
+                format("expected Iceberg %s(%s) bucket with %sd buckets to be %d, got %d", icebergType, value, Integer.MAX_VALUE, expectedMaxIntBucket, icebergBucketHash));
+
+        // Ensure hash is stable and does not change
+        assertEquals(
+                trinoBucketHash,
+                expectedMaxIntBucket,
+                format("expected Trino %s(%s) bucket with %sd buckets to be %d, got %d", icebergType, value, Integer.MAX_VALUE, expectedMaxIntBucket, trinoBucketHash));
+    }
+
+    private Integer computeIcebergBucket(Type type, Object value, int bucketCount)
+    {
+        Transform<Object, Integer> bucketTransform = Transforms.bucket(type, bucketCount);
+        assertTrue(bucketTransform.canTransform(type), format("bucket function %s is not able to transform type %s", bucketTransform, type));
+        return bucketTransform.apply(value);
+    }
+
+    private Integer computeTrinoBucket(Type icebergType, Object value, int bucketCount)
+    {
+        io.trino.spi.type.Type trinoType = toTrinoType(icebergType, TYPE_MANAGER);
+        Function<Block, Block> bucketTransform = getBucketTransform(trinoType, bucketCount);
+
+        BlockBuilder blockBuilder = trinoType.createBlockBuilder(null, 1);
+
+        if (value != null) {
+            if (trinoType == INTEGER) {
```

Comment:
What about `BIGINT` and date/time types? (the latter is mentioned in the TODO but `BIGINT` / `long` is not)

Reply:
here, we're converting from iceberg representation to trino representation.

for BIGINT, we don't need to convert, since `long` is on both ends.

we could make the conversion code exhaustive, as in `io.trino.plugin.hive.util.TestHiveBucketing#toNativeContainerValue`